scripts/ci/run-jenkins.sh

```diff
@@ -34,7 +34,17 @@ if [[ -n "${ghprbPullId}" ]] && [[ ${label} == w* ]];
     # we don't run the test suite on Windows PRs, we just ensure the build succeeds, so end here
 fi
 ${TESTCMD} --label=mini --timeout=5m make -w -C mono/mini -k check
+if [[ ${label} == w* ]]; then
+	# Create a temporary config which maps libtest to the MSVC built libtest.dll. Without this the P/Invoke tests in mono/tests/ won't find libtest.dll.
+	MONO_CONFIG=$(mktemp --suffix=.xml)
+	echo "<configuration><dllmap dll=\"libtest\" target=\"${WORKSPACE}\\msvc\\${PLATFORM}\\bin\\Release\\libtest.dll\"/></configuration>" > "$MONO_CONFIG"
+	export MONO_CONFIG=$(cygpath -w -a "$MONO_CONFIG")
+fi
 ${TESTCMD} --label=runtime --timeout=120m make -w -C mono/tests -k test-wrench V=1 CI=1
+if [[ ${label} == w* ]] && [[ -n "$MONO_CONFIG" ]] && [[ -f "$MONO_CONFIG" ]]; then
+	rm -f "$MONO_CONFIG"
+	unset MONO_CONFIG
+fi
```

Comment:
The problem with that is that `MONO_CONFIG` overrides the default config lookup so e.g. the configuration in `runtime/etc/mono/config` won't be read anymore which could have unintended side effects.

But... I just saw we basically already do the same in [Makefile.am#L23](https://github.com/mono/mono/blob/ab4737fce5b1b17bb727dcd74b098474238a45d6/mono/tests/Makefile.am#L23) which seems to be used only in the unmanaged test runner 😭 

So I guess we could try this out and see what happens 😄. I'd very much prefer if we didn't need the MONO_LIBTEST_PATH magic, can you please try if adding a dllmap path for the msvc library to tests-config.in fixes both windows and non-windows (i.e. it is ignored in the latter case)?